google-cloud-logging/src/test/java/com/google/cloud/logging/LoggingHandlerTest.java

```diff
@@ -381,6 +383,40 @@ public void testFlushLevel() {
   }
 
   @Test
+  public void testFlush() throws InterruptedException {
+    final SettableApiFuture<Void> mockRpc = SettableApiFuture.create();
+
+    EasyMock.expect(options.getProjectId()).andReturn(PROJECT).anyTimes();
+    EasyMock.expect(options.getService()).andReturn(logging);
+    logging.writeAsync(ImmutableList.of(INFO_ENTRY), DEFAULT_OPTIONS);
+    EasyMock.expectLastCall().andReturn(mockRpc);
+    EasyMock.replay(options, logging);
+    final LoggingHandler handler = new LoggingHandler(LOG_NAME, options);
+    handler.setFormatter(new TestFormatter());
+
+    // no messages, nothing to flush.
+    handler.flush();
+
+    // send a message
+    handler.publish(newLogRecord(Level.INFO, MESSAGE));
+    Thread flushWaiter = new Thread(new Runnable() {
+      @Override
+      public void run() {
+        handler.flush();
+      }
+    });
+    flushWaiter.start();
+
+    // flushWaiter should be waiting for mockRpc to complete.
+    flushWaiter.join(100);
+    assertTrue(flushWaiter.isAlive());
+
+    // With the RPC completed, flush should return, and the thread should terminate.
+    mockRpc.set(null);
+    flushWaiter.join();
```

Comment:
Maybe add a timeout so that the test doesn't lock up forever given an implementation bug.